python/pyspark/ml/classification.py

```diff
@@ -1560,8 +1581,9 @@ def trainSingleClass(index):
                             (classifier.predictionCol, predictionCol)])
             return classifier.fit(trainingDataset, paramMap)
 
-        # TODO: Parallel training for all classes.
-        models = [trainSingleClass(i) for i in range(numClasses)]
+        pool = ThreadPool(processes=min(self.getParallelism(), numClasses))
```

Comment:
In Scala, it doesn't matter because this just sets the max size of the pool and creates threads as needed, so it will never go above numClasses anyway.  In Python its a little unclear what it's doing, the `ThreadPool` class is not well documented, so I'm not sure if it makes a difference here.